src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandlerTest.java

```diff
@@ -1,136 +1,183 @@
+    @Test
+    void givenCaseCannotBeListedRightAwayAndNoDirectionsBeingMade_thenNewHearingRequestSent() {
+        PreSubmitCallbackResponse<SscsCaseData> response = cannotBeListedAndNoDirectionsGiven();
+
+        assertHearingCreatedAndAdjournmentInProgress(response);
+    }
+
+    @DisplayName("When adjournment is enabled and case is LA and case can be listed right away "
+        + "then should send a new hearing request in hearings API")
+    @Test
+    void givenCanBeListedRightAway_thenNewHearingRequestSent() {
+        PreSubmitCallbackResponse<SscsCaseData> response = canBeListed();
+
+        assertHearingCreatedAndAdjournmentInProgress(response);
+    }
+
+    private void assertHearingCreatedAndAdjournmentInProgress(PreSubmitCallbackResponse<SscsCaseData> response) {
+        verify(hearingMessageHelper, times(1))
+            .sendListAssistCreateHearingMessage(sscsCaseData.getCcdCaseId());
+
+        assertThat(response.getErrors()).isEmpty();
+        assertThat(response.getData().getAdjournment().getAdjournmentInProgress()).isEqualTo(YES);
+    }
+
+    @DisplayName("When adjournment is enabled and case is LA and case cannot be listed right away "
+        + "and directions are being made, then should not send any messages")
+    @Test
+    void givenCaseCannotBeListedRightAwayAndDirectionsAreBeingMade_thenNoMessagesSent() {
+        PreSubmitCallbackResponse<SscsCaseData> response = cannotBeListedAndDirectionsGiven();
+
+        verifyNoInteractions(hearingMessageHelper);
+
+        assertThat(response.getErrors()).isEmpty();
+    }
+
+    @DisplayName("When we have changed the next hearing venue through an adjournment, show we change the region")
+    @Test
+    void givenAdjournCaseNextHearingVenueSelectedTrue_thenSetRegion() {
+        String venueId = "185";
+
+        RegionalProcessingCenter rpc = getRegionalProcessingCenter();
+        String postcode = rpc.getPostcode();
+
+        DynamicListItem venue = new DynamicListItem(venueId, null);
+        DynamicList adjournedNextVenue = new DynamicList(venue, null);
+        BenefitType benefitType = BenefitType.builder().code("PIP").build();
+
+        sscsCaseData.getAdjournment().setNextHearingVenueSelected(adjournedNextVenue);
+        sscsCaseData.setAppeal(Appeal.builder()
+            .appellant(Appellant.builder()
+                .address(Address.builder().postcode(postcode).build()).isAppointee(YES.getValue())
+                .build())
+            .benefitType(benefitType)
+            .build());
+
+        when(airLookupService.lookupAirVenueNameByPostCode(postcode, benefitType)).thenReturn("cardiff");
+        when(regionalProcessingCenterService.getByVenueId(venueId)).thenReturn(rpc);
+
+        PreSubmitCallbackResponse<SscsCaseData> response =
+            handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+
+        assertThat(response.getErrors()).isEmpty();
+
+        assertThat(sscsCaseData.getRegion()).isEqualTo("CARDIFF");
```

Comment:
Does there need to be an assert earlier on in the test to show that getRegion() returns something other than "CARDIFF"? ie Make it clear that the region has changed. I am struggling to see where that is happening as this test is currently written.

Reply:
changed it to rpc.getCity()